packages/vue-components/src/Media.vue

```diff
@@ -0,0 +1,100 @@
+<template>
+  <div class="media-container">
+    <Pic
+      ref="content"
+      class="media-image"
+      :src="src"
+      :alt="alt"
+      :width="imageWidth"
+      :height="imageHeight"
+      :eager="eager"
+    />
+    <div class="content">
+      <slot></slot>
+    </div>
+  </div>
+</template>
+
+<script>
+import MarkdownIt from 'markdown-it';
+import Pic from './Pic.vue';
+
+const md = new MarkdownIt();
+
+export default {
+  components: {
+    Pic,
+  },
+  props: {
+    src: {
+      type: String,
+      default: '',
+    },
+    alt: {
+      type: String,
+      default: '',
+    },
+    imageWidth: {
+      type: String,
+      default: '100',
+    },
+    imageHeight: {
+      type: String,
+      default: '100',
+    },
+    eager: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  methods: {
+    processSlotText(text) {
+      let lines = text.split('\n');
+      while (lines.length > 0 && lines[0].trim() === '') {
+        lines.shift();
+      }
+      const firstLineIndent = lines[0].search(/\S/);
+      lines = lines.map((line) => {
+        const lineIndent = line.search(/\S/);
+        if (lineIndent < firstLineIndent) {
+          return line.trim();
+        }
+        return line.slice(firstLineIndent);
+      });
+      return lines.join('\n');
+    },
+  },
+  computed: {
+    isMarkdownText() {
+      const item = this.$slots.default ? this.$slots.default[0] : null;
+      if (!item || !item.text) {
+        return false;
+      }
+      return item.text.trim()[0] !== '<';
+    },
+  },
+  mounted() {
+    if (this.isMarkdownText) {
+      const processedText = this.processSlotText(this.$slots.default[0].text);
+      this.$refs.content.innerHTML = md.render(processedText);
```

Comment:
We used to do this long long ago but found several issues:
- FOUC, not good for UX. (that goes for any runtime heavy UI processing)
- Much higher bundle size (keeping in mind all the plugins we've added too)
- Rendering markdown twice on the back then frontend is undefined behaviour and very error prone, and difficult to ensure consistency.

Solution should be the same as above.

Reply:
I have some doubt...
1) FOUC (Flash of Unstyled Content) is primarily due to CSS that doesn't load prior to the HTML text. However, based on the component above, FOUC is unlikely to occur since it uses scoped styling and the Bootstrap CSS, which is the primary CSS used in the project.

2) Appreciate your input, I did overlook on this.

3) Presently, the component has been parsed into pure HTML by the Vue component on the server side within the Markbind context. It remains uncertain for me if it would be re-rendered as markdown on the client side again.

Reply:
\1. The most common cause is CSS by far but not limited to it. Some examples:
   - Dynamic style detection and adjustment on initialization (e.g. reading the user agent and adjust some styles based on it)
   - Manipulating the DOM into its "final" form after initialization (what you're doing here with `md.render`)

\3. `mounted()` hooks won't run on server side as it requires a DOM